frontend/src/metabase/components/form/CustomForm/CustomFormSubmit.tsx

```diff
@@ -22,12 +23,14 @@ function CustomFormSubmit({
   renderSubmit,
   disablePristineSubmit,
   children,
+  shouldPersistError,
+  fields,
   ...props
 }: CustomFormSubmitProps & FormLegacyContext) {
   const title = children || submitTitle || t`Submit`;
   const canSubmit = !(
     submitting ||
-    invalid ||
+    (shouldPersistError ? areAllFieldsUntouched(fields) : invalid) ||
```

Comment:
This way, I only change the form that set `shouldPersistError: true`'s behavior. Normally, after any field is visited the error messages would be cleared.
https://www.loom.com/share/8728395752504fc58adadac7c1e21130 

p.s. the logic is a bit confusing as everything combined is negated at the end. Actually, it should have read *Disable submit button when all fields are untouched (after form submission)*

Reply:
Note to self, this logic is invalid. I need to fix it.